bson/raw_value.go

```diff
@@ -260,6 +260,14 @@ func (rv RawValue) Int32() int32 { return convertToCoreValue(rv).Int32() }
 // panicking.
 func (rv RawValue) Int32OK() (int32, bool) { return convertToCoreValue(rv).Int32OK() }
 
+// AsInt32 returns a BSON number as an int32. If the BSON type is not a numeric one, this method
+// will panic.
+func (rv RawValue) AsInt32() int32 { return convertToCoreValue(rv).AsInt32() }
+
+// AsInt32OK is the same as AsInt32, except that it returns a boolean instead of
+// panicking.
+func (rv RawValue) AsInt32OK() (int32, bool) { return convertToCoreValue(rv).AsInt32OK() }
+
```

Comment:
Curious, do you know if this was explicitly left out because of the TODO on bsoncore Value? https://github.com/mongodb/mongo-go-driver/blob/c1d560769a77e8ff7a7b4cd8fe6ee288316a6d61/x/bsonx/bsoncore/value.go#L59-L64

Though IMO that TODO does not seem like a reason not to expose this. Adding support for converting Decimal128 later doesn't seem like a harmful change in behavior.

Reply:
I'm not actually sure why it wasn't originally included, but I couldn't think of a good reason for not to be added.